modules/apps/27-interchain-accounts/controller/ibc_module_test.go

```diff
@@ -632,3 +636,67 @@ func (suite *InterchainAccountsTestSuite) TestOnTimeoutPacket() {
 		})
 	}
 }
+
+func (suite *InterchainAccountsTestSuite) TestSingleHostMultipleControllers() {
+	var (
+		pathAToB *ibctesting.Path
+		pathCToB *ibctesting.Path
+	)
+
+	testCases := []struct {
+		msg      string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"success",
+			func() {},
+			true,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.msg, func() {
+			suite.SetupTest() // reset
+
+			// Setup a new path from A(controller) -> B(host)
+			pathAToB = NewICAPath(suite.chainA, suite.chainB)
+			suite.coordinator.SetupConnections(pathAToB)
+
+			err := SetupICAPath(pathAToB, TestOwnerAddress)
+			suite.Require().NoError(err)
+
+			// Setup a new path from C(controller) -> B(host)
+			// NOTE: Here the version metadata is overridden to include to the next host connection sequence (i.e. chainB's connection to chainC)
+			nextConnectionSeq := suite.chainB.App.GetIBCKeeper().ConnectionKeeper.GetNextConnectionSequence(suite.chainB.GetContext())
+			TestVersion = string(icatypes.ModuleCdc.MustMarshalJSON(&icatypes.Metadata{
+				Version:                icatypes.Version,
+				ControllerConnectionId: ibctesting.FirstConnectionID,
+				HostConnectionId:       connectiontypes.FormatConnectionIdentifier(nextConnectionSeq),
+			}))
+			pathCToB = NewICAPath(suite.chainC, suite.chainB)
+			suite.coordinator.SetupConnections(pathCToB)
```

Comment:
probably could have reordered this so you could use pathCToB.Endpoint for the connection sequence

Reply:
Yea nice catch, updated with your suggestion, thanks!